sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -1605,3 +1605,103 @@ private case class GetTimestamp(
   override def withTimeZone(timeZoneId: String): TimeZoneAwareExpression =
     copy(timeZoneId = Option(timeZoneId))
 }
+
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(year, month, day, hour, min, sec, timezone) - Create timestamp from year, month, day, hour, min, sec and timezone fields.",
+  arguments = """
+    Arguments:
+      * year - the year to represent, from 1 to 9999
+      * month - the month-of-year to represent, from 1 (January) to 12 (December)
+      * day - the day-of-month to represent, from 1 to 31
+      * hour - the hour-of-day to represent, from 0 to 23
+      * minute - the minute-of-hour to represent, from 0 to 59
+      * second  the second-of-minute and its micro-fraction to represent, from
+      *         0 to 59.999999
+  """,
+  examples = """
+    Examples:
+      > SELECT _FUNC_(2014, 12, 28, 6, 30, 45.887);
+       2014-12-28 06:30:45.887
+      > SELECT _FUNC_(2019, 13, 1, 10, 11, 12, 13);
+       NULL
+      > SELECT _FUNC_(null, 7, 22, 15, 30, 0);
+       NULL
+  """,
+  since = "3.0.0")
+// scalastyle:on line.size.limit
+case class MakeTimestamp(
+    year: Expression,
+    month: Expression,
+    day: Expression,
+    hour: Expression,
+    min: Expression,
+    sec: Expression,
+    timeZoneId: Option[String] = None)
+  extends SenaryExpression with TimeZoneAwareExpression with ImplicitCastInputTypes {
```

Comment:
PostgreSQL `make_timestamp` return `timestamp without time zone`. Can we have a description for this difference?


postgres=# select make_timestamp(1,1,1,0,0,60);
   make_timestamp
---------------------
 0001-01-01 00:01:00
(1 row)

Reply:
This ins not specific to the function. Spark's `TIMESTAMP` is similar to Oracle's `TIMESTAMP WITH LOCAL TIME ZONE`: https://docs.oracle.com/cd/B28359_01/server.111/b28298/ch4datetime.htm#i1006169

Reply:
I updated description of the PR regarding the `TIMESTAMP` type.